src/ds/d500/d500-auto-calibration-handler.h

```diff
@@ -0,0 +1,66 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include "d500-private.h"
+#include <src/hw-monitor.h>
+
+
+namespace librealsense
+{
+    enum class d500_calibration_state
+    {
+        RS2_D500_CALIBRATION_STATE_IDLE = 0,
+        RS2_D500_CALIBRATION_STATE_PROCESS,
+        RS2_D500_CALIBRATION_STATE_DONE_SUCCESS,
+        RS2_D500_CALIBRATION_STATE_DONE_FAILURE,
+        RS2_D500_CALIBRATION_STATE_FLASH_UPDATE,
+        RS2_D500_CALIBRATION_STATE_COMPLETE
+    };
+
+    enum class d500_calibration_result
+    {
+        RS2_D500_CALIBRATION_RESULT_UNKNOWN = 0,
+        RS2_D500_CALIBRATION_RESULT_SUCCESS,
+        RS2_D500_CALIBRATION_RESULT_FAILED_TO_CONVERGE,
+        RS2_D500_CALIBRATION_RESULT_FAILED_TO_RUN
+    };
+
+    enum class d500_calibration_mode
+    {
+        RS2_D500_CALIBRATION_MODE_RESERVED = 0,
+        RS2_D500_CALIBRATION_MODE_RUN,
+        RS2_D500_CALIBRATION_MODE_ABORT,
+        RS2_D500_CALIBRATION_MODE_DRY_RUN
+    };
+
+#pragma pack(push, 1)
+    struct d500_calibration_answer
+    {
+        uint8_t calibration_state;
+        int8_t calibration_progress;
+        uint8_t calibration_result;
+        ds::d500_coefficients_table depth_calibration;
+    };
+#pragma pack(pop)
+
+    class d500_device;
+    class debug_interface;
+
+    class d500_auto_calibrated_handler
+    {
```

Comment:
Also, specifically here, this looks like a `d500_triggered, calibration_handler`. Not relevant for other calibrations like FL.

Reply:
> A question regarding naming convention. The file is called `calibration` but the class `calibrated`. Should be the same?

Agree with you - this was done this way for d400

> Also, specifically here, this looks like a `d500_triggered, calibration_handler`. Not relevant for other calibrations like FL.

correct - some renaming must be done here - the thing is that I do not want to use the word "triggered" because it is not informative at all. Any idea?

Reply:
"Triggered" isn't informative same as "on chip calibration". Both don't give you a clue as to what the calibration actually does. Just that the user activates it or that it is done in FW.
But this is the name whole other R&D teams are using so it might be confusing to give it another name.

If you know what the calibration fixes you can use it - `triggered_ppx_calibration` etc...

Reply:
changed the methods name with triggered_calibration, as I do not see any better idea